src/canaries/include/canaries/Workloads.hpp

```diff
@@ -0,0 +1,200 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_521F5952_C79A_407E_9348_661B514D288D_INCLUDED
+#define HEADER_521F5952_C79A_407E_9348_661B514D288D_INCLUDED
+
+#include <chrono>
+#include <thread>
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+
+#include <mongocxx/client.hpp>
+
+#include <gennylib/v1/ConfigNode.hpp>
+#include <gennylib/v1/PoolManager.hpp>
+
+namespace genny::canaries {
+
+class Task {
+public:
+    virtual inline void run() = 0;
+
+protected:
+    template <class T>
+    inline void doNotOptimize(T const& value) {
+        asm volatile("" : : "r,m"(value) : "memory");
```

Comment:
🤯 is this..documented somewhere? or copied from somewhere?

Reply:
I added a comment. It's adapted from Google Benchmark's namesake function.